lib/DBIx/Class/ParseError/Parser.pm

```diff
@@ -12,6 +11,30 @@ has _schema => (
     is => 'ro', required => 1, init_arg => 'schema',
 );
 
+has custom_errors => (
+    is      => 'ro',
+    default => sub { {} },
+);
+
+# Feels weird putting the BUILD method in a role, but this effectively acts as
+# a base class. We can't use a method modifier because BUILD isn't in the
+# inheritance hierarchy of the classes and I didn't think it was appropriate
+# to change too much.
```

Comment:
Yeah, right, it was something I've thought during initial dev/design and ended up deciding for a role for reasons I don't remember now. Open to change that to a base class.